stan/math/prim/fun/ordered_free.hpp

```diff
@@ -23,23 +23,23 @@ namespace math {
  * @throw std::domain_error if y is not a vector of positive,
  *   ordered scalars.
  */
-template <typename T>
-Eigen::Matrix<T, Eigen::Dynamic, 1> ordered_free(
-    const Eigen::Matrix<T, Eigen::Dynamic, 1>& y) {
+template <typename EigVec, require_eigen_col_vector_t<EigVec>* = nullptr>
+Eigen::Matrix<value_type_t<EigVec>, Eigen::Dynamic, 1> ordered_free(
+    const EigVec& y) {
   check_ordered("stan::math::ordered_free", "Ordered variable", y);
   using Eigen::Dynamic;
   using Eigen::Matrix;
   using std::log;
-  using size_type = index_type_t<Matrix<T, Dynamic, 1>>;
+  using size_type = Eigen::Index;
 
   size_type k = y.size();
-  Matrix<T, Dynamic, 1> x(k);
+  Eigen::Matrix<value_type_t<EigVec>, Eigen::Dynamic, 1> x(k);
```

Comment:
Same thing here about plain object and auto

Reply:
fixed